src/modules/function/invocation_utils.rs

```diff
@@ -28,11 +28,14 @@ fn ordinal_number(index: usize) -> String {
 fn run_function_with_args(meta: &mut ParserMetadata, mut fun: FunctionDecl, args: &[Type], tok: Option<Token>) -> Result<(Type, usize), Failure> {
     // Check if there are the correct amount of arguments
     if fun.arg_names.len() != args.len() {
+        let max_args = fun.arg_names.len();
+        let min_args = fun.arg_names.len() - fun.arg_optionals.len();
+        let opt_argument = if max_args > min_args {&format!(" ({} optional)",max_args)} else {""};
```

Comment:
doesn't compile:

```
error[E0716]: temporary value dropped while borrowed
  --> src/modules/function/invocation_utils.rs:33:53
   |
33 |         let opt_argument = if max_args > min_args {&format!(" ({} optional)",max_args)} else {""};
   |             ------------                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- temporary value is freed at the end of this statement
   |             |                                       |
   |             borrow later stored here                creates a temporary value which is freed while still in use
   |
   = note: consider using a `let` binding to create a longer lived value
   = note: this error originates in the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0716`.
error: could not compile `amber` (bin "amber") due to 1 previous error
```

idk why the CI build is running smoothly

Reply:
maybe a different rust version?

Reply:
> rustc 1.80.0-nightly

you shouldn't use nightly if you are not a rust dev/tester. even though it has some features that regular rust doesnt have, its unstable and shouldnt be used for production puropses

Reply:
I'm on ``` rustc 1.79.0 ``` and it compiles. Which rust version are you using?

Reply:
I tested it on rust 1.79.0 on linux and works.
@b1ek what is the version you are using?

Reply:
> you shouldn't use nightly if you are not a rust dev/tester. even though it has some features that regular rust doesnt have, its unstable and shouldnt be used for production puropses

Yep, I was testing something before on the nightly build